OpenRA.Mods.Common/UpdateRules/Rules/RemoveExplodesEmptyWeapon.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+    public class SplitExplodes : UpdateRule
+    {
+        public override string Name { get { return "Remove Explodes EmptyWeapon parameters"; } }
+        public override string Description
+        {
+            get
+            {
+                return "Explodes.EmptyWeapon has been removed as it became obsolete by introducing conditions.\n" +
+                       "Using conditions allows to specify more precisely when and which Explodes should be used.";
+            }
+        }
+
+        readonly List<Tuple<string, string>> emptyWeaponLocations = new List<Tuple<string, string>>();
+
+        public override IEnumerable<string> AfterUpdate(ModData modData)
+        {
+            var message1 = "EmptyWeapon has been removed from Explodes.\n"
+                           + "Instead, use a second Explodes, and select the correct one using the RequiresCondition property.\n"
+                           + "If you were using the LoadedChance property, grant a condition by using GrantRandomCondition.\n"
+                           + "The following actors have been updated and might need manual adjustments:\n"
+                           + UpdateUtils.FormatMessageList(emptyWeaponLocations.Select(n => n.Item1 + " (" + n.Item2 + ")"));
+
+            if (emptyWeaponLocations.Any())
+                yield return message1;
+
+            emptyWeaponLocations.Clear();
+        }
+
+        public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+        {
+            foreach (var explodes in actorNode.ChildrenMatching("Explodes"))
+            {
+                var weapon = explodes.LastChildMatching("Weapon");
+                var emptyWeapon = explodes.LastChildMatching("EmptyWeapon");
+                var loadedChance = explodes.LastChildMatching("LoadedChance");
+
+                if (weapon == null && emptyWeapon == null)
+                    explodes.AddNode("Weapon", "UnitExplode");
```

Comment:
This looks dangerous. What if you are overwriting an inherited value that is not `UnitExplode`? (Inheritance is the only way `weapon` can be `null` here anyway, I think, as the field has `FieldLoader.Require`.)
The same for `emptyWeapon.Key = "Weapon";` below.

Reply:
> The following actors have been updated and might need manual adjustments

IMO the current behaviour (doing something that works but might be wrong) is better than generating an incomplete trait definiton (missing a required field) that crashes.